src/watts_ui/watts_ui.py

```diff
@@ -0,0 +1,610 @@
+        elif param.bool is not None:
+            bool_str = str(param.bool.value).strip('\"')
+            if bool_str.upper() == "TRUE":
+                params[param_id] = True
+            else:
+                params[param_id] = False
+
+        params_id.append(str(param.id))
+
+    return (params)
+
+
+def convert_to_list(wb_list):
+    """Convert a Workbench list to Python list
+
+    Parameters
+    ----------
+    wb_list
+        Workbench list to convert
+
+    Returns
+    -------
+    convert_list
+        Converted Python list
+
+    """
+    convert_list = []
+    for n, val in enumerate(wb_list.value):
+        string = str(wb_list.value[n]).strip('\"')
+        if isfloat(string):
+            convert_list.append(float(string))
+        else:
+            convert_list.append(string)
+    return (convert_list)
+
+
+def get_last_value(watts_params, name_list):
+    """Extract the value of the last index from the
+    results created by WATTS plugins for iteration workflow
+
+    Parameters
+    ----------
+    watts_params
+        WATTS params
+    name_list
+        List of name of params to extract values
+
+    Returns
+    -------
+    watts_params
+        Updated WATTS params
+
+    """
+    for param_name in name_list:
+        if isinstance(watts_params[param_name], (list, np.ndarray)):
+            watts_params[param_name] = watts_params[param_name][-1]
+    return (watts_params)
+
+
+def isfloat(num):
+    """Check whether entries of a list are numeric
+
+    Parameters
+    ----------
+    num
+        Entry of list
+
+    Returns
+    -------
+    True
+        If entry is numeric
+
+    """
+    try:
+        float(num)
+        return True
+    except ValueError:
+        return False
+
+
+def run_workflow(watts_params, wf_level, plugin_ID, watts_plugins):
+    """Run workflow
+
+    Parameters
+    ----------
+    watts_params
+        Watts params with stored user input parameters
+    wf_level
+        Level of workflow
+    plugin_ID
+        ID of plugin
+    plugin
+        Dictionary of plugin
+
+    Returns
+    -------
+    app_result
+        WATTS results
+
+    """
+    if wf_level.iteration is not None:
+        watts_params, app_result_plugin_1, app_result_plugin_2 = run_iterate(
+            watts_params,  watts_plugins, wf_level)
+
+        # Combine the results from plugins 1 and 2 into a dictionary
+        app_result = {'app_result_plugin_1': app_result_plugin_1,
+                      'app_result_plugin_2': app_result_plugin_1}
+
+        return (app_result, watts_params)
+
+    elif wf_level.parametric is not None:
+        watts_params, app_result = run_parametric(
+            watts_params,  watts_plugins[plugin_ID], wf_level)
+
+        return (app_result, watts_params)
+
+    elif wf_level.optimization is not None:
+        operation = wf_level.optimization
+        ...
+    else:
+        watts_params, app_result = run_direct(
+            watts_params, watts_plugins[plugin_ID])
+
```

Comment:
in the run_direct - there may be ambiguity on the orders of the different plugins. I am not sure Workbench will return you the plugin in a given order. We may ask a user to provide some ordering for the workflow to clarify the order the different plugins needs execution such as: `plugin(1)=ID1`

Reply:
The code now determines the sequence of plugin based on the order. For example, if we have

plugin = ID1
plugin = ID2

The code will run ID1 first then ID2.